File: Configurations/Differential/ggH/plot/TDRStyle.cc
#include "TGaxis.h"
#include "TStyle.h"
#include "TLatex.h"

void TDRStyle(){
  ///===========================================================================
  
  //  gROOT->Reset();
  //  gROOT->SetStyle("Plain");
  //  gStyle->SetPalette(1); 
  //  gStyle->SetOptStat(1110);
  //  gStyle->SetPadTickX(1);
  //  gStyle->SetPadTickY(1);
  //  gStyle->SetTitleBorderSize(0);
  
  // style:
  
  //For the canvas:
  gStyle->SetCanvasBorderMode(0);
  gStyle->SetCanvasColor(kWhite);
  gStyle->SetCanvasDefH(800); //600
  //Height of canvas
  gStyle->SetCanvasDefW(800);//800 //Width of canvas
  gStyle->SetCanvasDefX(0);   //POsition on screen
  gStyle->SetCanvasDefY(0);
  
  // For the Pad:
  gStyle->SetPadBorderMode(0);
  // gStyle->SetPadBorderSize(Width_t size = 1);
  gStyle->SetPadColor(kWhite);
  gStyle->SetPadGridX(false);
  gStyle->SetPadGridY(false);
  gStyle->SetGridColor(0);
  gStyle->SetGridStyle(3);
  gStyle->SetGridWidth(1);
  
  //For the frame:
  gStyle->SetFrameBorderMode(0);
  gStyle->SetFrameBorderSize(1);
  gStyle->SetFrameFillColor(0);
  gStyle->SetFrameFillStyle(0);
  gStyle->SetFrameLineColor(1);
  gStyle->SetFrameLineStyle(1);
  gStyle->SetFrameLineWidth(1);
  
  // For the histo:
  gStyle->SetHistLineColor(1);
  gStyle->SetHistLineStyle(0);
  gStyle->SetHistLineWidth(2);
  gStyle->SetEndErrorSize(2);
  //  gStyle->SetErrorMarker(20);
  //gStyle->SetErrorX(0.);
  gStyle->SetMarkerStyle(8);
  
  
  // For the legend:
  //  gStyle->SetFillColor(8);
  gStyle->SetFillStyle(0);
  
  
  
  // For the statistics box:
  gStyle->SetOptFile(0);
  //  gStyle->SetOptStat(1);
  gStyle->SetOptStat(0);
  gStyle->SetStatColor(kWhite);
  gStyle->SetStatFont(42);
  gStyle->SetStatFontSize(0.025);
  gStyle->SetStatTextColor(1);
  gStyle->SetStatFormat("6.4g");
  gStyle->SetStatBorderSize(1);
  //   gStyle->SetStatBorderSize(0);
  gStyle->SetStatH(.1);
  gStyle->SetStatW(.15);
  
  //  gStyle->SetStatX(.9);
  // gStyle->SetStatY(.9);
  
  // For the Global title:
  gStyle->SetOptTitle(0);
  //   gStyle->SetOptTitle(1);
  gStyle->SetTitleFont(42);
  gStyle->SetTitleColor(1);
  gStyle->SetTitleTextColor(1);
  gStyle->SetTitleFillColor(10);
  //  gStyle->SetTitleFontSize(0.05);
  gStyle->SetTitleFontSize(0.04);
  gStyle->SetTitleX(0.4);
  
  // For the axis titles:
  gStyle->SetTitleColor(1, "XYZ");
  gStyle->SetTitleFont(42, "XYZ");
  //  gStyle->SetTitleSize(0.06, "XYZ");
  gStyle->SetTitleSize(0.04, "XYZ");
  gStyle->SetTitleXOffset(0.9);
  // gStyle->SetTitleYOffset(1.25);
  //gStyle->SetTitleXOffset(0.5);
  gStyle->SetTitleYOffset(1.0);
  
  // For the axis labels:
  gStyle->SetLabelColor(1, "XYZ");
  gStyle->SetLabelFont(42, "XYZ");
  gStyle->SetLabelOffset(0.007, "XYZ");
  
  gStyle->SetLabelSize(0.05, "XYZ");
  //  gStyle->SetLabelSize(0.02, "XYZ");
  
  // For the axis:
  gStyle->SetAxisColor(1, "XYZ");
  gStyle->SetStripDecimals(kTRUE);
  gStyle->SetTickLength(0.03, "XYZ");
  gStyle->SetNdivisions(510, "XYZ");
  gStyle->SetPadTickX(0);  // To get tick marks on the opposite side of the frame
  gStyle->SetPadTickY(1);
  // // Postscript options:
  //   //gStyle->SetPaperSize(20.,20.);
  
  // CC style  
  gStyle->SetTitleXOffset(0.8);
  //   gStyle->SetTitleYOffset(0.8);
  gStyle->SetTitleYOffset(1.0);
  
  gStyle->SetLabelOffset(0.005, "XYZ");
  gStyle->SetTitleSize(0.07, "XYZ");
  gStyle->SetTitleFont(22,"X");
  gStyle->SetTitleFont(22,"Y");
  /*  gStyle->SetPadBottomMargin(0.13);
   * gStyle->SetPadTopMargin(0.23);
   * gStyle->SetPadLeftMargin(0.15);
   * gStyle->SetPadRightMargin(0.21);*/
  /*  //PlotPt4Lep=====================
   * gStyle->SetPadBottomMargin(0.15);
   * gStyle->SetPadTopMargin(0.05);
   * gStyle->SetPadLeftMargin(0.15);
   * gStyle->SetPadRightMargin(0.05);*/
  /*  //PlotPrePreSelInvMass=====================
   * gStyle->SetPadBottomMargin(0.15);
   * gStyle->SetPadTopMargin(0.10);
   * gStyle->SetPadLeftMargin(0.15);
   * gStyle->SetPadRightMargin(0.10);*/
  /*  //PlotPrePreSelEff=====================
   * gStyle->SetPadBottomMargin(0.15);
   * gStyle->SetPadTopMargin(0.15);
   * gStyle->SetPadLeftMargin(0.15);
   * gStyle->SetPadRightMargin(0.15);*/
  //PlotInvMass=====================
  gStyle->SetPadBottomMargin(0.17);
  gStyle->SetPadTopMargin(0.07);
  //  gStyle->SetPadTopMargin(0.12);
  gStyle->SetPadLeftMargin(0.17);
  gStyle->SetPadRightMargin(0.12);
  /*  //PlotEffvsEff=====================
   * gStyle->SetPadBottomMargin(0.15);
   * gStyle->SetPadTopMargin(0.05);
   * gStyle->SetPadLeftMargin(0.15);
   * gStyle->SetPadRightMargin(0.05);*/
  /*  //JetMET=====================
   * gStyle->SetPadBottomMargin(0.15);
   * gStyle->SetPadTopMargin(0.05);
   * gStyle->SetPadLeftMargin(0.15);
   * gStyle->SetPadRightMargin(0.05);*/
  //   //Met2dEff=====================
  //   gStyle->SetPadBottomMargin(0.15);
  //   gStyle->SetPadTopMargin(0.05);
  //   gStyle->SetPadLeftMargin(0.15);
  //   gStyle->SetPadRightMargin(0.15);
  //   //Dalitz+PtVsMET=====================
  //   gStyle->SetPadBottomMargin(0.15);
  //   gStyle->SetPadTopMargin(0.10);
  //   gStyle->SetPadLeftMargin(0.15);
  //   gStyle->SetPadRightMargin(0.15);
  
  /*  gStyle->SetPadBottomMargin(0.13);
   * gStyle->SetPadLeftMargin(0.15);*/
  //  gStyle->SetHistLineWidth(3);
  gStyle->SetHistLineWidth(2);
  gStyle->SetLineStyleString(5,"20 12 4 12");
  gStyle->SetLineStyleString(6,"20 12 4 12 4 12 4 12");
  gStyle->SetLineStyleString(7,"20 20");
  gStyle->SetLineStyleString(8,"20 12 4 12 4 12");
  gStyle->SetLineStyleString(9,"80 20");
  
  
  
  
  
  TGaxis::SetMaxDigits(2);
  
  
  gStyle->cd(); 
  
  ///===========================================================================
}

File: Configurations/EFT/ggH/Full2018/nAODv7_Full2018v7_preApprovalV02/tdrstyle.py
import ROOT as rt

def tdrGrid( gridOn):
  tdrStyle.SetPadGridX(gridOn)
  tdrStyle.SetPadGridY(gridOn)

#fixOverlay: Redraws the axis
def fixOverlay(): gPad.RedrawAxis()

def setTDRStyle():
  tdrStyle =  rt.TStyle("tdrStyle","Style for P-TDR")

   #for the canvas:
  tdrStyle.SetCanvasBorderMode(0)
  tdrStyle.SetCanvasColor(rt.kWhite)
  tdrStyle.SetCanvasDefH(600) #Height of canvas
  tdrStyle.SetCanvasDefW(600) #Width of canvas
  tdrStyle.SetCanvasDefX(0)   #POsition on screen
  tdrStyle.SetCanvasDefY(0)


  tdrStyle.SetPadBorderMode(0)
  #tdrStyle.SetPadBorderSize(Width_t size = 1)
  tdrStyle.SetPadColor(rt.kWhite)
  tdrStyle.SetPadGridX(False)
  tdrStyle.SetPadGridY(False)
  tdrStyle.SetGridColor(0)
  tdrStyle.SetGridStyle(3)
  tdrStyle.SetGridWidth(1)

#For the frame:
  tdrStyle.SetFrameBorderMode(0)
  tdrStyle.SetFrameBorderSize(1)
  tdrStyle.SetFrameFillColor(0)
  tdrStyle.SetFrameFillStyle(0)
  tdrStyle.SetFrameLineColor(1)
  tdrStyle.SetFrameLineStyle(1)
  tdrStyle.SetFrameLineWidth(1)
  
#For the histo:
  #tdrStyle.SetHistFillColor(1)
  #tdrStyle.SetHistFillStyle(0)
  #tdrStyle.SetHistLineColor(1)
  #tdrStyle.SetHistLineStyle(0)
  #tdrStyle.SetHistLineWidth(1)
  #tdrStyle.SetLegoInnerR(Float_t rad = 0.5)
  tdrStyle.SetNumberContours(50)

  tdrStyle.SetEndErrorSize(2)
  #tdrStyle.SetErrorMarker(20)
  #tdrStyle.SetErrorX(0.)
  
  tdrStyle.SetMarkerStyle(20)
  
#For the fit/function:
  tdrStyle.SetOptFit(1)
  tdrStyle.SetFitFormat("5.4g")
  tdrStyle.SetFuncColor(2)
  tdrStyle.SetFuncStyle(1)
  tdrStyle.SetFuncWidth(1)

#For the date:
  tdrStyle.SetOptDate(0)
  # tdrStyle.SetDateX(Float_t x = 0.01)
  # tdrStyle.SetDateY(Float_t y = 0.01)

# For the statistics box:
  tdrStyle.SetOptFile(0)
  tdrStyle.SetOptStat(0) # To display the mean and RMS:   SetOptStat("mr")
  tdrStyle.SetStatColor(rt.kWhite)
  tdrStyle.SetStatFont(42)
  tdrStyle.SetStatFontSize(0.025)
  tdrStyle.SetStatTextColor(1)
  tdrStyle.SetStatFormat("6.4g")
  tdrStyle.SetStatBorderSize(1)
  tdrStyle.SetStatH(0.1)
  tdrStyle.SetStatW(0.15)
  # tdrStyle.SetStatStyle(Style_t style = 1001)
  # tdrStyle.SetStatX(Float_t x = 0)
  # tdrStyle.SetStatY(Float_t y = 0)

# Margins:
  tdrStyle.SetPadTopMargin(0.06) #0.05
  tdrStyle.SetPadBottomMargin(0.13)
  tdrStyle.SetPadLeftMargin(0.14)
  tdrStyle.SetPadRightMargin(0.05) #0.14 for 2D

# For the Global title:

  tdrStyle.SetOptTitle(0)
  tdrStyle.SetTitleFont(42)
  tdrStyle.SetTitleColor(1)
  tdrStyle.SetTitleTextColor(1)
  tdrStyle.SetTitleFillColor(10)
  tdrStyle.SetTitleFontSize(0.05)
  # tdrStyle.SetTitleH(0) # Set the height of the title box
  # tdrStyle.SetTitleW(0) # Set the width of the title box
  # tdrStyle.SetTitleX(0) # Set the position of the title box
  # tdrStyle.SetTitleY(0.985) # Set the position of the title box
  # tdrStyle.SetTitleStyle(Style_t style = 1001)
  # tdrStyle.SetTitleBorderSize(2)

# For the axis titles:

  tdrStyle.SetTitleColor(1, "XYZ")
#  tdrStyle.SetTitleFont(48, "XYZ")
#  tdrStyle.SetTitleSize(0.10, "XYZ")
  tdrStyle.SetTitleXSize(0.045) # Another way to set the size?
  tdrStyle.SetTitleYSize(0.055)
  tdrStyle.SetTitleXOffset(1.25)
  tdrStyle.SetTitleYOffset(1.90)
  # tdrStyle.SetTitleOffset(1.1, "Y") # Another way to set the Offset

# For the axis labels:

  tdrStyle.SetLabelColor(1, "XYZ")
  tdrStyle.SetLabelFont(42, "XYZ")
  tdrStyle.SetLabelOffset(0.007, "XYZ")
  tdrStyle.SetLabelSize(0.05, "XYZ")

# For the axis:

  tdrStyle.SetAxisColor(1, "XYZ")
  tdrStyle.SetStripDecimals(True)
  tdrStyle.SetTickLength(0.03, "XYZ")
  tdrStyle.SetNdivisions(510, "XYZ")
  tdrStyle.SetPadTickX(1)  # To get tick marks on the opposite side of the frame
  tdrStyle.SetPadTickY(1)

# Change for log plots:
  tdrStyle.SetOptLogx(0)
  tdrStyle.SetOptLogy(0)
  tdrStyle.SetOptLogz(0)

# Postscript options:
  tdrStyle.SetPaperSize(20.,20.)
  # tdrStyle.SetLineScalePS(Float_t scale = 3)
  # tdrStyle.SetLineStyleString(Int_t i, const char* text)
  # tdrStyle.SetHeaderPS(const char* header)
  # tdrStyle.SetTitlePS(const char* pstitle)

  # tdrStyle.SetBarOffset(Float_t baroff = 0.5)
  # tdrStyle.SetBarWidth(Float_t barwidth = 0.5)
  # tdrStyle.SetPaintTextFormat(const char* format = "g")
  # tdrStyle.SetPalette(Int_t ncolors = 0, Int_t* colors = 0)
  # tdrStyle.SetTimeOffset(Double_t toffset)
  # tdrStyle.SetHistMinimumZero(kTRUE)

  tdrStyle.SetHatchesLineWidth(5)
  tdrStyle.SetHatchesSpacing(0.05)

  tdrStyle.cd()

  return tdrStyle
